src/olympia/promoted/models.py

```diff
@@ -0,0 +1,57 @@
+from django.db import models
+
+from olympia.addons.models import Addon
+from olympia.amo.models import ModelBase
+from olympia.constants.promoted import (
+    NOT_PROMOTED, PROMOTED_GROUPS, PROMOTED_GROUPS_BY_ID)
+from olympia.versions.models import Version
+
+
+class PromotedAddon(ModelBase):
+    GROUP_CHOICES = [(group.id, group.name) for group in PROMOTED_GROUPS]
+    group_id = models.SmallIntegerField(
+        choices=GROUP_CHOICES, default=NOT_PROMOTED.id)
+    addon = models.OneToOneField(
+        Addon, on_delete=models.CASCADE, null=False)
+
+    @property
+    def group(self):
+        return PROMOTED_GROUPS_BY_ID.get(self.group_id, NOT_PROMOTED)
+
+    def __getattr__(self, name):
+        # magic to return the fixed properties of the promoted group too
+        try:
+            super().__getattribute__(name)
+        except AttributeError as parent_attr_error:
+            try:
+                return getattr(self.group, name)
+            except AttributeError:
+                raise parent_attr_error
+
+    def is_addon_promoted(self, addon=None):
```

Comment:
I like the overall strategy of how the models are set up but I am not a big fan of this API: Fetching a `PromotedAddon` only to have to call `is_addon_promoted()` to check feels weird. It also doesn't really tell you that the `current_version` is what really matters...

Also, I guess being able to pass `addon` helps avoid extra queries but that makes the API even weirder...

Reply:
Implementing it like `Addon.is_recommended` is the alternative, but it places the logic in the Addon model instead and I was trying to avoid that again.  Maybe I should make it a classmethod that handles fetching the `PromotedAddon` too?

Reply:
So actually... when django instantiates the related instance from the `onetoonefield` it should automatically set the parent instance, avoiding extra queries (said differently: `addon.promotedaddon.addon` should already automatically be the same instance than `addon`, avoiding extra queries). So, I'd keep this as a regular method, but remove the ability to pass an `addon`. And maybe rename to `is_addon_current_version_promoted` or `is_addon_currently_promoted` or something...

Reply:
>  Maybe I should make it a classmethod that handles fetching the PromotedAddon too?

/me tries that

Reply:
https://github.com/mozilla/addons-server/pull/14898/commits/714cae363455aba4f8edf98457e145e8ed18654a